user-impl/src/main/java/com/example/auction/user/impl/CredentialEntity.java

```diff
@@ -0,0 +1,45 @@
+package com.example.auction.user.impl;
+
+import com.lightbend.lagom.javadsl.persistence.PersistentEntity;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public class CredentialEntity extends PersistentEntity<CredentialCommand, CredentialEvent, Optional<PCredential>> {
```

Comment:
I'm not sure the reason for separating the credential entity out from the user entity - perhaps I wasn't clear enough before. I meant that update credentials should be a new event in the user entity, not that there should be two different entities. I think if you copy `UpdateCredential` into `UserCommand`, and `CredentialUpdated` into `UserEvent`, and then the behaviour in this class can be copied into the `UserEntity`, that will achieve what I was thinking.